src/main/java/it/rate/webapp/controllers/InterestController.java

```diff
@@ -32,12 +32,19 @@ public class InterestController {
 
   @PreAuthorize("hasAnyAuthority('USER', 'ADMIN')")
   @GetMapping("/create")
-  public String createPage(Model model) {
+  public String createPage(Model model, Principal principal) {
     List<Criterion> criteria = new ArrayList<>();
     model.addAttribute("criteria", criteria);
     model.addAttribute("interest", new Interest());
     model.addAttribute("action", "/interests/create");
     model.addAttribute("method", "post");
+    if (principal != null) {
+      model.addAttribute(
+          "loggedUser",
+          userService
+              .findByEmail(principal.getName())
+              .orElseThrow(() -> new RuntimeException("Email not found in the database")));

```

Comment:
also here, use getByEmail method instead of findByEmail. In this case, it should be a 100% that the user is found.